### PR TITLE
Adapt all tests for executing in mobile browser (Android)

### PR DESCRIPTION
**Pull request by akhivyk**

### CODE REVIEW

**Review by Trukhanovich — APPROVED** [review]

File: src/main/java/com/solvd/carina/demo/gui/mobilworld/android/BlogPage.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.solvd.carina.demo.gui.mobilworld.android;
+
+import com.solvd.carina.demo.gui.mobilworld.common.BlogPageBase;
+import com.zebrunner.carina.utils.factory.DeviceType;
+import org.openqa.selenium.WebDriver;
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = BlogPage.class)
```
Comment (jjanickamatraszek) [inline src/main/java/com/solvd/carina/demo/gui/mobilworld/android/BlogPage.java:7]: parentClass should be BlogPageBase.class instead of BlogPage.class

File: src/main/java/com/solvd/carina/demo/gui/mobilworld/android/IphonePage.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public boolean isModelsPresent() {
+        return models.isEmpty();
+    }
+
+    @Override
+    public ExtendedWebElement findModel(String model) {
+        return models.stream()
+                .filter(e -> e.getText().equals(model))
+                .findFirst()
+                .orElse(null);
```
Comment (jjanickamatraszek) [inline src/main/java/com/solvd/carina/demo/gui/mobilworld/android/IphonePage.java:39]: instead of returning null when model is not found you can throw an exception that will describe the problem using orElseThrow();

File: src/main/java/com/solvd/carina/demo/gui/mobilworld/android/MainPage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        setPageAbsoluteURL("https://mobilworld.by/");
+    }
+
+    public Menu getMenu() {
+        return menu;
+    }
+
+    @Override
+    public DevicesPage openDevicePage() {
+        return null;
```
Comment (jjanickamatraszek) [inline src/main/java/com/solvd/carina/demo/gui/mobilworld/android/MainPage.java:27]: in carina-demo project you can see that when method is not supported by for example Android then instead of return null there is throw new UnsupportedOperationException(THIS_METHOD_IS_DEFINED_ONLY_IN_DESKTOP);

and private static final String THIS_METHOD_IS_DEFINED_ONLY_IN_DESKTOP = "This method is not yet implemented for Android" is defined at the beginning of the class;

File: src/main/java/com/solvd/carina/demo/gui/mobilworld/android/SecondIphonePage.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        waitUntil(ExpectedConditions.visibilityOfAllElementsLocatedBy(By.xpath("//a[contains(@href, '/catalog/iphone')]")), 10);
+        return iphoneModels.isEmpty();
+    }
+
+    @Override
+    public ExtendedWebElement findModel(String model) {
+        return iphoneModels.stream()
+                .filter(e -> e.getText().equals(model))
+                .findFirst()
+                .orElse(null);
```
Comment (jjanickamatraszek) [inline src/main/java/com/solvd/carina/demo/gui/mobilworld/android/SecondIphonePage.java:34]: just like before you can throw an exception when model is not present

File: src/main/java/com/solvd/carina/demo/gui/mobilworld/android/SecondIphonePage.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public ItemPage openItemPage(String model) {
+        ExtendedWebElement element = findModel(model);
+        if (element != null) {
+            element.clickByJs();
+            System.out.println("ENTERED");
+            return new ItemPage(driver);
+        }
+        return null;
```
Comment (jjanickamatraszek) [inline src/main/java/com/solvd/carina/demo/gui/mobilworld/android/SecondIphonePage.java:44]: 1) if you throw exception then you dont need to check if element is null;
2) you can remove  System.out.println("ENTERED");

File: src/main/resources/_config.properties [hunk 6]
```diff
@@ -39,13 +39,14 @@ init_retry_interval=1
...
-#capabilities.deviceType=phone
-#capabilities.platformName=ANDROID
-#capabilities.app=https://qaprosoft.s3-us-west-2.amazonaws.com/carinademoexample.apk
+selenium_url=http://localhost:4723/wd/hub
+capabilities.automationName=uiautomator2
+capabilities.deviceName=Pixel 6 API 33
+capabilities.udid=emulator-5554
+capabilities.deviceType=phone
+capabilities.platformName=ANDROID
+capabilities.app=/Users/akhivyk/Downloads/com.android.chrome.apk
```
Comment (jjanickamatraszek) [inline src/main/resources/_config.properties:48]: you dont need to attach chrome apk to test webside, as chrome is natively installed on Android

File: src/main/resources/_config.properties [hunk 7]
```diff
@@ -39,13 +39,14 @@ init_retry_interval=1
...
+capabilities.appWaitPackage=D://Dev//Solvd//instagram.apk
```
Comment (jjanickamatraszek) [inline src/main/resources/_config.properties:49]: why did you add this instagram apk?

File: src/test/java/com/solvd/carina/demo/MobileMWTest.java [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class MobileMWTest implements IAbstractTest {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    @Test
+    public void testLogin() {
+        LoginPage loginPage = new LoginPage(getDriver());
```
Comment (jjanickamatraszek) [inline src/test/java/com/solvd/carina/demo/MobileMWTest.java:19]: I think that in tests you should operate on Base classes, so instead of LoginPage you should use LoginPageBase.
As there should be a single set of tests for all platforms. 
And now you have tests only for android with imports
import com.solvd.carina.demo.gui.mobilworld.android.*;
import com.solvd.carina.demo.gui.mobilworld.android.components.Menu;

**Review by jjanickamatraszek — COMMENTED** [review]
some suggestions
